data/src/error_analysis/erroranalysis.py

```diff
@@ -0,0 +1,70 @@
+"""Error Analysis tool for G2P.
+
+This script assumes 2 input files. a) Covering Grammer b) Test output. 
+CG file  : Each line contain graphime and their corresponding pronunciation seperated by a tab
+Test File: contains three attributes in each line seperated by a tab. The attributes are - orthography, expected pronunciation, and hypothesized pronunciation."""
+
+__author__ = "Arundhati Sengupta"
+
+import pynini
+import argparse
+
+def main(args: argparse.Namespace):
+    cg_fst = pynini.string_file(args.cg_path).closure().optimize()
+    rulematch_predMatch = 0
+    rulematch_predNotMatch = 0
+    notRulematch_predMatch = 0
+    notRulematch_predNotMatch = 0
+    total_records = 0
+    with open(args.test_path, "r") as source:
+        for line in source:
+            total_records+=1
+            parts = line.split('\t')
+            lg = parts[0].strip()
+            act = parts[1].replace(' ','').replace('.','').strip()
+            predPron = parts[2].replace(' ','').replace('.','').strip()
+             
+            lattice = (lg @ cg_fst @ predPron).project(True)
+            
+            if lattice.start() == pynini.NO_STATE_ID:
+                if (act == predPron):
+                    notRulematch_predMatch += 1
+                    
+                else:
+                    notRulematch_predNotMatch += 1            
+            else:
+                if (act == predPron):
+                    rulematch_predMatch += 1
+                else:
+                    rulematch_predNotMatch += 1
+    
+    print ('Total Number of Records', total_records)
+    ruleMpredNM = round(rulematch_predNotMatch / total_records, 4) * 100
```

Comment:
Camel-case, also don't round these. Instead do ```rule_m_pred_nm = 100 * rulematch_pred_nm / total``` and use the table-writing function to round instead.

Reply:
What I understand from this comment that the table-writing function will take two parameters and return the percentage value. Please let me know if understood it incorrectly.

Reply:
Not exactly. It doesn't matter too much whether the table-writing function takes as arguments the counts you use to compute the percentages, or the percentages. What does matter is that it you don't round before you need to. It's better to use the string-formatting procedures which round for you than to round explicitly, and so you could pass `rule_m_pred_nm`, unrounded, to the table-writing function.

Reply:
That's a good point. If we do not round the value and use the string format directly on the decimal values, the cumulative value reported in the table may not be 100%. This is because string format usages truncation. Therefore, first, we are round up the values into 4 decimal points and then using string format to make sure summing all the cell values will result in 100%.

Reply:
I don't think that's right, string formatting rounds too:

```
In [1]: f"{2/3:.1f}"
Out[1]: '0.7'
```

Reply:
sorry, my mistake. I have overdone it. I will fix this issue. Thank you.

Reply:
fixed.